pyo3-build-config/src/impl_.rs

```diff
@@ -1924,6 +2058,51 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_not_cross_compiling_from_to() {
```

Comment:
Maybe also add a case where it should cross-compile?

Reply:
That requires setting `PYO3_CROSS_LIB_DIR` env var to something, and I don't know how to do that in a unit test.
https://github.com/PyO3/pyo3/pull/2241 lifts this restriction, so is becomes much easier to add there.